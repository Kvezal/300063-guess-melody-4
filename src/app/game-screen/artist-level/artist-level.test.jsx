import React from "react";
import render from "react-test-renderer";

import ArtistLevel from "./artist-level";


describe(`ArtistLevel`, () => {
  const question = {
    type: `artist`,
    song: {artist: `Jim Beam`, src: `src`},
    answers: [
      {picture: `picture/1`, artist: `John Snow`},
      {picture: `picture/2`, artist: `Jack Daniels`},
      {picture: `picture/3`, artist: `Jim Beam`}
    ],
  };

  test(`should render component`, () => {
    const tree = render
      .create(
          <ArtistLevel
            question={question}
            onAnswer={() => {}}
            renderPlayer={() => {}}
          />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
