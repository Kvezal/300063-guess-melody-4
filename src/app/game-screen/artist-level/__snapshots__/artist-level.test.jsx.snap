// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistLevel should render component 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="John Snow"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="John Snow"
      />
      <label
        className="artist__name"
        htmlFor="John Snow"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="picture/1"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="Jack Daniels"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="Jack Daniels"
      />
      <label
        className="artist__name"
        htmlFor="Jack Daniels"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="picture/2"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="Jim Beam"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="Jim Beam"
      />
      <label
        className="artist__name"
        htmlFor="Jim Beam"
      >
        <img
          alt="Jim Beam"
          className="artist__picture"
          src="picture/3"
        />
        Jim Beam
      </label>
    </div>
  </form>
</section>
`;
